File: examples/databases/db-grant-group.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Grant an entire group manager-level access to a database.
export async function grantDatabaseToGroup(client: JsBaoClient, databaseId: string) {
  // #region example
  await client.databases.grantGroupPermission(databaseId, {
    groupType: "team",
    groupId: "engineering",
    permission: "manager",
  });
  // #endregion example
}

File: examples/databases/db-grant-group.swift
import JsBaoClient

// Grant an entire group manager-level access to a database.
func grantDatabaseToGroup(client: JsBaoClient, databaseId: String) async throws {
  // #region example
  _ = try await client.databases.grantGroupPermission(
    databaseId: databaseId,
    params: ["groupType": "team", "groupId": "engineering", "permission": "manager"]
  )
  // #endregion example
}
